--- origsrc/fluidsynth-1.1.5/configure.ac	2011-09-04 02:38:58.000000000 -0500
+++ src/fluidsynth-1.1.5/configure.ac	2011-10-23 19:22:47.857759600 -0500
@@ -64,11 +64,17 @@ LIBFLUID_LIBS="-lm"
 
 dnl Machine specific checks and defines
 case "${host_os}" in
+cygwin*)
+  winmidi_support="yes"
+  LIBFLUID_LDFLAGS="-no-undefined"
+  LIBFLUID_LIBS="-lwinmm"
+  ;;
 darwin*)
   AC_DEFINE(DARWIN, 1, [Define if building for Mac OS X Darwin])
   ;;
 mingw*)
   mingw32_support="yes"
+  winmidi_support="yes"
   CFLAGS="$CFLAGS -mms-bitfields"
   AC_DEFINE(MINGW32, 1, [Define if using the MinGW32 environment])
   LIBFLUID_LIBS="-ldsound -lwinmm -lws2_32"
@@ -95,6 +101,7 @@ AC_LIB_PROG_LD_GNU
 AM_CONDITIONAL(GNU_LD_SUPPORT, test "x$with_gnu_ld" = "xyes")
 
 AM_CONDITIONAL(MINGW32_SUPPORT, test "$mingw32_support" == "yes")
+AM_CONDITIONAL(WINMIDI_SUPPORT, test "$winmidi_support" == "yes")
 
 AC_ARG_ENABLE(double, AS_HELP_STRING([--enable-double],
 	[double floating point for dsp (default=float)]),
--- origsrc/fluidsynth-1.1.5/src/Makefile.am	2011-09-04 02:38:58.000000000 -0500
+++ src/fluidsynth-1.1.5/src/Makefile.am	2011-10-23 19:23:48.245464900 -0500
@@ -33,7 +33,11 @@ fluid_portaudio = drivers/fluid_portaudi
 endif
 
 if MINGW32_SUPPORT
-fluid_windows = fluid_dll.c drivers/fluid_dsound.c drivers/fluid_winmidi.c
+fluid_windows = fluid_dll.c drivers/fluid_dsound.c
+endif
+
+if WINMIDI_SUPPORT
+fluid_winmidi = drivers/fluid_winmidi.c
 endif
 
 if OSS_SUPPORT
@@ -94,6 +98,7 @@ libfluidsynth_la_SOURCES = \
 	$(fluid_portaudio) \
 	$(fluid_pulse) \
 	$(fluid_windows) \
+	$(fluid_winmidi) \
 	$(fluid_dart) \
 	$(fluid_rtkit) \
     utils/fluid_conv.c \
--- origsrc/fluidsynth-1.1.5/src/drivers/fluid_winmidi.c	2011-09-04 02:38:58.000000000 -0500
+++ src/fluidsynth-1.1.5/src/drivers/fluid_winmidi.c	2011-10-23 20:11:41.446074700 -0500
@@ -37,6 +37,8 @@
 #include "fluid_midi.h"
 #include "fluid_mdriver.h"
 #include "fluid_settings.h"
+#include <windows.h>
+#include <mmsystem.h>
 
 #define MIDI_SYSEX_MAX_SIZE     512
 #define MIDI_SYSEX_BUF_COUNT    16
--- origsrc/fluidsynth-1.1.5/src/utils/fluidsynth_priv.h	2011-09-04 02:38:58.000000000 -0500
+++ src/fluidsynth-1.1.5/src/utils/fluidsynth_priv.h	2011-10-23 20:10:39.732367100 -0500
@@ -140,6 +140,10 @@
 #define WITHOUT_SERVER 1
 #endif
 
+#ifdef __CYGWIN__
+#define WINMIDI_SUPPORT 1
+#endif
+
 
 #include "fluidsynth.h"
 
